### PR TITLE
Stage Losses Update

### PR DESCRIPTION
**Pull request by owenmylotte**
Moves the radiation losses to the pre-stage function to provide time stepping stability. The radiation gains are evaluated at the interval. This allows the volumetric radiation to run stable with mass fraction dependent absorption.

### CODE REVIEW

**Review by owenmylotte — COMMENTED** [review]
The only substantial change is that the calls to the solve and the radiation gain evaluation are separated.
All of the current tests pass, but the optimized build segfault is still present in the slab burner. I'll try running some additional simple cases to isolate it. (I've gotten the source down to the sublimation solver absorption function call at this point, maybe an initialization issue). It may be worth adding additional integration tests for different radiation and absorption classes to test that all of the solver interactions are working?
Once the optimized build issue is figured out then I'm confident we have a running full radiation slab burner.

File: src/boundarySolver/physics/sublimation.cpp [hunk 1]
```diff
@@ -295,15 +301,16 @@ void ablate::boundarySolver::physics::Sublimation::Setup(PetscInt numberSpeciesI
...
-PetscErrorCode ablate::boundarySolver::physics::Sublimation::SublimationPreStep(BoundarySolver &solver, TS ts, PetscReal time, bool initialStage, Vec locX, void *ctx) {
+PetscErrorCode ablate::boundarySolver::physics::Sublimation::SublimationPreRHS(BoundarySolver &solver, TS ts, PetscReal time, bool initialStage, Vec locX, void *ctx) {
     PetscFunctionBegin;
     auto sublimation = (Sublimation *)ctx;
     PetscInt step;
     TSGetStepNumber(ts, &step) >> checkError;
     TSGetTime(ts, &time) >> checkError;
     if (initialStage && sublimation->radiationInterval->Check(PetscObjectComm((PetscObject)ts), step, time)) {
-        sublimation->radiation->Solve(solver.GetSubDomain().GetSolutionVector(), solver.GetSubDomain().GetField("temperature"), solver.GetSubDomain().GetAuxVector());
+        sublimation->radiation->EvaluateGains(solver.GetSubDomain().GetSolutionVector(), solver.GetSubDomain().GetField("temperature"), solver.GetSubDomain().GetAuxVector());
```
Comment (mmcgurn) [inline src/boundarySolver/physics/sublimation.cpp:311]: This is a good name change.

File: src/eos/radiationProperties/zimmer.cpp [hunk 2]
```diff
@@ -198,6 +220,8 @@ ablate::eos::ThermodynamicTemperatureFunction ablate::eos::radiationProperties::
                                     .densityYiCO2Offset = CO2offset,
                                     .densityYiCOOffset = COoffset,
                                     .densityYiCH4Offset = CH4offset,
+                                    .eulerOffset = eulerField->offset,
```
Comment (mmcgurn) [inline src/eos/radiationProperties/zimmer.cpp:223]: Where do you use this Euler offset?

**Review by mmcgurn — APPROVED** [review]
I have a question about the added eulerOffset to the zimmer model.

**Pull request merged**